instat/sdgClimaticDataEntryOptions.vb

```
@@ -106,8 +105,7 @@ Public Class sdgClimaticDataEntryOptions
 
     Public ReadOnly Property TransformValue As Double
         Get
-            'todo. do explicit conversion. Currently ucrInputTransform validation type is number
-            Return If(String.IsNullOrEmpty(ucrInputTransform.GetValue), 0, ucrInputTransform.GetValue)
+            Return ucrInputTransform.GetValue
```

Would it improve fault tolerance if we did a `TryCast` here to ensure that the `GetValue` object can safely be cast to a double?

---

I think you meant `TryParse`? If this is correct then I agree with you.

---

@lloyddewit I thought of changing this into a function as follows;
```
    Public Function TransformValue() As Object
        Dim dTemp As Double
        If Double.TryParse(ucrInputTransform.GetValue, dTemp) Then
            Return ucrInputTransform.GetValue
        Else
            Return CDbl(ucrInputTransform.GetValue)
        End If
    End Function
```
What do you think?

---

Yes, you're right `TryParse` is the correct function' I liked your function. For me, keeping it as a readonly property is also fine. I think if the `TryParse` fails then we should raise a developer error rather than trying `CDbl`. How about something like below?
```
    Public ReadOnly Property GetTransformValue As Double
        Get
            Dim dTemp As Double
            If Not Double.TryParse(ucrInputTransform.GetValue, dTemp) Then 'If TryParse fails then sets dTemp to zero
                MsgBox("Developer error: The ucrInputTransform combo box contains the value '" & ucrInputTransform.GetValue & "'. Cannot convert this to a double.")
            End If
            Return dTemp 
        End Get
    End Property
```

---

Agreed